aspnetcore/security/authentication/policyschemes.md

```diff
@@ -0,0 +1,91 @@
+        /// For example Context.ForbidAsync("ThisScheme") => Context.ForbidAsync("ForwardForbidValue");
+        /// Set the target to the current scheme to disable forwarding and allow normal processing.
+        /// </summary>
+        public string ForwardForbid { get; set; }
+
+        /// <summary>
+        /// If set, this specifies the target scheme that this scheme should forward SignInAsync calls to.
+        /// For example Context.SignInAsync("ThisScheme") => Context.SignInAsync("ForwardSignInValue");
+        /// Set the target to the current scheme to disable forwarding and allow normal processing.
+        /// </summary>
+        public string ForwardSignIn { get; set; }
+
+        /// <summary>
+        /// If set, this specifies the target scheme that this scheme should forward SignOutAsync calls to.
+        /// For example Context.SignOutAsync("ThisScheme") => Context.SignInAsync("ForwardSignOutValue");
+        /// Set the target to the current scheme to disable forwarding and allow normal processing.
+        /// </summary>
+        public string ForwardSignOut { get; set; }
+
+        /// <summary>
+        /// Used to select a default scheme for the current request that authentication handlers should forward all authentication operations to
+        /// by default. The default forwarding logic will check the most specific ForwardAuthenticate/Challenge/Forbid/SignIn/SignOut 
+        /// setting first, followed by checking the ForwardDefaultSelector, followed by ForwardDefault. The first non null result
+        /// will be used as the target scheme to forward to.
+        /// </summary>
+        public Func<HttpContext, string> ForwardDefaultSelector { get; set; }
+    }
+```
+
+
+
+## Examples
+* A higher level scheme that combines lower level schemes, where Google is used for challenges, and Cookie is used for everything else.
+
+```C#
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                .AddCookie(options => options.ForwardChallenge = "Google")
+                .AddGoogle(options => { });
+        }
+```
+
+* Enables dynamic selection of schemes on a per request basis (i.e. how to mix cookies and api authentication)
+
+```C#
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                .AddCookie(options =>
+                {
+                    // For example, can foward any requests that start with /api to the api scheme
+                    options.ForwardDefaultSelector = ctx =>
+                    {
+                        return ctx.Request.Path.StartsWithSegments("/api") ? "Api" : null;
+                    })
+                    .AddYourApiAuth("Api");
```

Comment:
The formatting here is off, you're missing `; }` no?

Reply:
I'm moving all the embedded snippets to a file I can compile and check. I'll create a PR into this PR.

Reply:
Cool, some of this is pseudocode though, like `AddYourApiAuth`